### PR TITLE
OpenApi 3.0 support to mimic v2 usage style #393

### PR DESCRIPTION
**Pull request by balajiv113**
The following channges are done,
* Moved common configurations, properties, builder and models to **swagger2Markup-core** module
* Extended **swagger2Markup-core** module in **swagger2Markup-v2** and **swagger2Markup-v3** modules
* Provided MarkupComponent to create re-usable components for swagger2Markup-v3 (Like in previous version)

**swagger2Markup-core module**
* This module will be free of any swagger dependency.
* The entry point for any new swagger version will be writing a class extending **AbstractSchema2MarkupConverter**
* Write swagger version specific classes by extending **Schema2MarkupConfigBuilder**, **Labels**, **Schema2MarkupExtensionRegistry**, **Schema2MarkupProperties** if required. 
Note: Its better to extend these classes as it becomes future proof and any version specific changes can be done in the new class

**swagger2Markup-v2 module**

Code has been refactored to use the **swagger2Markup-core**. Existing test cases are verified and running properly (except instagram testcase which is failing even before my patch - in windows)
**Note:**
There are few breaking changes but are very trivial to fix like,
* Labels has been renamed to SwaggerLabels
* MarkupLanguage package has been modified
* Extensions uses io.github.swagger2markup.Swagger2MarkupConverter.SwaggerContext instead of io.github.swagger2markup.Swagger2MarkupConverter.Context

**swagger2Markup-v3 module**
* OpenAPI specific converter is created **io.github.swagger2markup.OpenAPI2MarkupConverter.**
* Extensions support similar to **swagger2Markup-v2** the only difference is instead of a markUpBuilder in v3 it accepts Document object.
* Refractored applyOverviewDocument to support the use of extensions and position

**Todo**

* Refractor openAPI-3 static classes similar to OverviewDocument with placeholders for calling extenstion points
* Move labels from OpenAPIHelpers to OpenAPILabels class
* Create re-usage MarkupComponent for all static methods under OpenAPIHelpers
* Current code doesn't provide a generic extension mechanism, it provides a marker interface **Schema2MarkupExtensionRegistry**. This should be changed and core should be able to give a proper extension mechanism

### CODE REVIEW

**Comment by RobWin** [comment]
Changes look good to me.
We can adjust some minore namings later. But it's a good base to go on.

**Comment by RobWin** [comment]
@balajiv113 If you would like to participate more in the refactoring, I can invite you to our Slack channel.-

**Comment by balajiv113** [comment]
@RobWin I would like to.

**Comment by RobWin** [comment]
Done. I sent you an invitiation

File: swagger2markup-core/build.gradle [hunk 1]
```diff
@@ -0,0 +1,25 @@
+ext.moduleName="io.github.swagger2markup.swagger-core"
```
Comment (austek) [inline swagger2markup-core/build.gradle]: module name should be `io.github.swagger2markup.core`

**Pull request merged**